### PR TITLE
Update payment method selection UI and fee display

### LINKED ISSUE #3300: Update UPE payment method selection and fees
Update the payment method selection and fees per paJDYF-2rk-p2

This update makes each payment method selectable from the main settings screen. 

Screen Shot 2021-11-05 at 8 41 33 AM

The pill that displays the fees should show the aggregate total of the fees that the merchant will pay for each payment method. Hovering over the pill will provide a breakdown of the components of the fee. 

Screen Shot 2021-11-05 at 8 41 20 AM

### PR DESCRIPTION
**Pull request by tpaksu**
Fixes #3300
Reference paJDYF-2rk-p2

### Changes proposed in this Pull Request

This PR applies the changes described in paJDYF-2rk-p2 to the UPE workflow:

## All screens
- Shows detailed fee breakdown for the PM
- Adds a 1500ms delay and loading animation to the checkbox check event
- Displays activation modal after checkbox delay if the PM has requirements

## Payment settings
- Removes the payment method selector modal and adds checkboxes directly to the payment method list
- Removes the delete button and moves the action to the checkbox uncheck event.
- Delete modal shows for disabling an already active PM
- Moves the activation modal display to the checkbox check event.
- Lists all the available methods instead showing only the enabled methods
- Removes the add payment method button

## UPE onboarding & UPE setup screens
- All the checkboxes start unchecked now, to display the activation modal when any requirements are present for the PM.

---

## Payment settings (http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments)

image

image

image

## UPE Onboarding (http://localhost:8082/wp-admin/admin.php?page=wc-admin&task=woocommerce-payments--additional-payment-methods)

image

image

## UPE Setup (http://localhost:8082/wp-admin/admin.php?page=wc-admin&task=woocommerce-payments--additional-payment-methods)

image

image

---

### Testing instructions

* All tests should pass
* Check if the screens are in sync with the flow described in paJDYF-2rk-p2 (and it's comments)
* You can check how to simulate different statuses for PM's in the testing instructions of  https://github.com/Automattic/woocommerce-payments/pull/3093 
* You can reach to the PM setup screen via returning this setting `false`: https://github.com/Automattic/woocommerce-payments/blob/de28a8d26f0a93507700fdaae78b18f2d8d8f6b8/includes/admin/class-wc-payments-admin.php#L326

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by kalessil** [comment]
Hey @tpaksu, curious about ETA for this PR. Any estimates for it (asking as we have 2 PR which marked as blocked by this PR)?

**Comment by kalessil** [comment]
Awesome, thanks for the prompt response @tpaksu !

**Comment by dmallory42** [comment]
Thanks for working on this! The flows work as described, I'll now take a look at the code before approving. I also think we should add GlobalStep testing flows for this, if we haven't already. :) 

I noticed a few minor UX things while testing the settings page:

1. When you select a new payment method, and then click `Cancel` on the confirmation modal, the payment method still stays selected. I think it should be unselected in these cases.

2. The `Pending Activation` pillboxes don't appear immediately after enabling a PM which requires additional verification. I needed to `Save Changes` and then refresh the page to see it. Is that expected?

3. This is more of a UX question, but I didn't find it clear that I needed to scroll down and click `Save Changes` to persist the changes I was making to the settings. Is there any way we can make this more obvious? (this is outside of the scope of the PR, I think).

File: client/utils/account-fees.js [hunk 1]
```diff
@@ -5,24 +5,172 @@
...
+	PL: 12,
+	PT: 13,
+	SG: 14,
+	ES: 15,
+	CH: 16,
+	UK: 17,
+	US: 18,
+};
+
+const getStripeFeeSectionUrl = ( country ) => {
```
Comment (dmallory42) [inline client/utils/account-fees.js:40]: Should we add logic to return a generic link to the page if the country isn't in our `countryFeeStripeDocsSectionNumbers` list? Just thinking we may forget to update this if we launch in new countries 😅

**Review by dmallory42 — APPROVED** [review]
LGTM! I think we should take care of the comment I mentioned about Cancelling a modal not behaving as expected, but apart from that everything looks good and works well! Left a few other comments for consideration but nothing blocking.

**Comment by tpaksu** [comment]
Hi @dmallory42 , thanks for reviewing this, I assume it was hard to read all this code, so I owe you a drink :) 

> When you select a new payment method, and then click Cancel on the confirmation modal, the payment method still stays selected. I think it should be unselected in these cases.
Yep, I reproduced it, and now I'm trying to solve it. Thanks for noticing it.

> The Pending Activation pillboxes don't appear immediately after enabling a PM which requires additional verification. I needed to Save Changes and then refresh the page to see it. Is that expected?
We get those statuses from the Stripe API, and until the next API call, which I believe will be triggered by the capability request, and then the `account.updated` webhook, the server cache will get the update, but I don't know about how the client will respond to that update. I didn't want to put the pill immediately, because after the refresh, it'll lose that one, because of the cached Stripe account info.

> This is more of a UX question, but I didn't find it clear that I needed to scroll down and click Save Changes to persist the changes I was making to the settings. Is there any way we can make this more obvious? (this is outside of the scope of the PR, I think).
I asked this a lot to David and Jarek, with giving the MC settings page as an example, and they told me everytime that MC screen is not working correctly, it should be having a "Save changes" button, so after I heard that, I stopped asking them about it :) 

I think after I fix the checkbox issue, it'll be good to ship, right?

**Comment by dmallory42** [comment]
Hey @tpaksu! No problem at all, the code follows a pretty clear pattern so it wasn't too tough to read! 

> Yep, I reproduced it, and now I'm trying to solve it. Thanks for noticing it.

Awesome! 

> We get those statuses from the Stripe API, and until the next API call, which I believe will be triggered by the capability request, and then the account.updated webhook, the server cache will get the update, but I don't know about how the client will respond to that update. I didn't want to put the pill immediately, because after the refresh, it'll lose that one, because of the cached Stripe account info.

That makes sense, thanks for explaining!

> I think after I fix the checkbox issue, it'll be good to ship, right?

I think so! I've preapproved it to ship once the checkbox issue is fixed but happy to test again if needed.

File: client/utils/account-fees.js (hunk 1 above)
Comment (tpaksu) [reply client/utils/account-fees.js:40]: Makes sense. I'll add a generic link directing the merchant to https://woocommerce.com/document/payments/faq/fees/ if the country doesn't exist in that list. :+1:

Comment (tpaksu) [reply client/utils/account-fees.js:46]: Solved in 94ede4f. Thanks for the suggestion :)

**Pull request merged**